lab-kyle/lib/note-crud.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const Note = require('../model/note');
+const AppErr = require('../lib/app-error');
+const debug = require('debug')('awesomeNote:note-crud');
+const userCrud = require('./user-crud');
+
+exports.createNote = function(reqBody){
+  debug('note-crud createNote');
+  return new Promise((resolve, reject) => {
+    if (! reqBody.content)
+      return reject(AppErr.error400('note require\'s content'));
+    if(! reqBody.name)
+      return reject(AppErr.error400('note require\'s name'));
+
+    reqBody.timestamp = new Date();
+    const note = new Note(reqBody);
+    note.save()
+    .then(resolve)
+    .catch(reject);
+  });
+};
+
+exports.fetchNoteByUserId = function(userId){
+  debug('note-crud fetchNoteByUserId');
+  return new Promise((resolve, reject) => {
+    userCrud.fetchUser({_id: userId})
```

Comment:
Not sure what you're doing here. Looks like you are looking up the user to by their id to get their id to use in finding an array of notes. Just use their id to look up the notes. And for consistency, do what you did above:

.then(resolve).catch(reject);